packages/flutter/lib/src/widgets/animated_raster.dart

```diff
@@ -0,0 +1,215 @@
+    animation.addListener(markNeedsPaint);
+    animation.addStatusListener(_updateStatus);
+    _updateStatus(animation.status);
+  }
+
+  AnimationStatus _status = AnimationStatus.completed;
+
+  void _updateStatus(AnimationStatus newStatus, [bool painting = false]) {
+    if (newStatus == _status) {
+      return;
+    }
+    _childImage?.dispose();
+    _childImage = null;
+    _status = newStatus;
+    assert(_status == animation.status);
+    if (!painting) {
+      markNeedsPaint();
+    }
+  }
+
+  /// The delegate that controls how the child raster is painted.
+  AnimatedRasterDelegate get delegate => _delegate;
+  AnimatedRasterDelegate _delegate;
+  set delegate(AnimatedRasterDelegate value) {
+    if (value == delegate) {
+      return;
+    }
+    _delegate = value;
+    markNeedsPaint();
+  }
+
+  /// The device pixel ratio used to create the child raster.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+
+  /// The animation used to drive this render object.
+  Animation<double> get animation => _animation;
+  Animation<double> _animation;
+  set animation(Animation<double> value) {
+    if (value == animation) {
+      return;
+    }
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    _animation = value;
+    animation.addStatusListener(_updateStatus);
+    animation.addListener(markNeedsPaint);
+    _updateStatus(animation.status);
+    markNeedsPaint();
+  }
+
+  @override
+  void dispose() {
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    _childImage?.dispose();
+    _childImage = null;
+    super.dispose();
+  }
+
+  @override
+  void attach(covariant PipelineOwner owner) {
+    animation.addListener(markNeedsPaint);
+    animation.addStatusListener(_updateStatus);
+    _updateStatus(animation.status);
+    super.attach(owner);
+  }
+
+  @override
+  void detach() {
+    animation.removeListener(markNeedsPaint);
+    animation.removeStatusListener(_updateStatus);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => child != null;
+
+  ui.Image? _childImage;
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child == null) {
+      return;
+    }
+    _updateStatus(animation.status, true);
+    switch (_status) {
+      case AnimationStatus.dismissed:
+      case AnimationStatus.completed:
+        super.paint(context, offset);
+        return;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        break;
+    }
+    final bool updateImage = delegate.willPaint(animation);
+    if (_childImage == null && updateImage) {
+      _childImage = PaintingContext.paintAndDetachToGpuImage(
```

Comment:
Will we have trouble if the children is changing during animation? For example, there may be a loading indicator that is animating during this zoom page transition. Or there may be a splash effect happening because user has just tapped a button.

In those cases, the content (i.e. childImage) actually should be updated, but we are caching it and never update it except when animation status changes.

Therefore, users may see the loading indicator or splash "freezed" during this zoom IMHO.

Reply:
That is a price I am willing to pay

Reply:
I agree. But maybe we should have some flags allowing users to customize it? Or maybe some documentation saying this is expected behavior - or people may file a bug about it

Reply:
@goderbauer did we ever change route transitions to turn the ticker mode off, or was that just in customer: money's app?

Reply:
I suspect that would not have improved raster performance at all for zoom page, given that the scale transform would invalidate all raster caches

Reply:
It was for the opacity one. Animation was busting the raster caching there

Reply:
> did we ever change route transitions to turn the ticker mode off, or was that just in customer: money's app?

We did not change TIckerModes with regards to transitions anywhere. We changed that changes in TickerMode don't cause unnecessary rebuilds in - mostly hidden - routes.

Reply:
> That is a price I am willing to pay

I am going back and forth on this. On the one hand, this is also how iOS handles transitions, I believe. On the other hand, this was a cool feature of flutter that we could do animations while transitioning...

At the very least, we need to document this behavior somewhere. I also wonder if we need a flag to opt out of this behavior?

One example where you absolutely want animations to run during transitions are Hero animations. It's the whole point of a Hero that they fly around while you transition between routes. How does this change affect Heros? (Maybe they are unaffected, because the Hero is placed in a separate Overlay?)

Also, there are transitions where when you trigger them manually (e.g. by swiping back) things actually animate on the screen (e.g. a back-arrow animates back to a Hamburger Menu button). Are any of these use cases impacted?